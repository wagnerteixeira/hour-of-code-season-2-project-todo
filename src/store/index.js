import { combineReducers, createStore, applyMiddleware } from 'redux';
import thunk from 'redux-thunk';

import layoutReducer from '../layout/redux/layoutReducer';
import todoReducer from '../todo/redux/todoReducer';

const reducers = combineReducers({  
  layout: layoutReducer,
  todo: todoReducer
});

export default createStore(reducers, applyMiddleware(thunk));
