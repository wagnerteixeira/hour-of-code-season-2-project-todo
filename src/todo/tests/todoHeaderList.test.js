import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import toJson from 'enzyme-to-json';
import { Provider } from 'react-redux';
import store from '../../store';

import TodoHeaderList from '../todoHeaderList';
import { todoMock } from './todoMock';

configure({ adapter: new Adapter() });

describe('Test <TodoHeaderList />', function () {

    let component;
    let goTo = jest.fn();
    beforeEach(() => {
        component = shallow(
            <TodoHeaderList
                title={todoMock.defaultHeaderTitle}
                status={todoMock.defaultStatus}
                goTo={goTo}
            />
        );
    });

    it('render <TodoHeaderList /> ', () => {
        expect(toJson(component)).toMatchSnapshot();
    });

});