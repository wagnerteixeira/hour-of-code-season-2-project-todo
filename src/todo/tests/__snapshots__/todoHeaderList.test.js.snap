// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test <TodoHeaderList /> render <TodoHeaderList />  1`] = `
<ContextConsumer>
  <Component />
</ContextConsumer>
`;
